### ISSUE TITLE
issue upgrading chado 1.2 to 1.3

### ISSUE THREAD

**Issue by srobb1** [tripal/tripal#970]
Hello,

I am expanding on a question I asked [here](https://github.com/NAL-i5K/T2-to-T3-Upgrade-Guide/issues/1)

My original question and response from @bradfordcondon:

----
We we try to upgrade from chado 1.2 to 1.3. When we are done, Tripal still tells up we have 1.2.

We tried your [premigration script](https://github.com/NAL-i5K/T2-to-T3-Upgrade-Guide/blob/master/pre_migration.sh), then try to upgrade chado again and we get this error

```
Calling: tripal_core_install_chado(Upgrade Chado v1.2 to v1.3, 3106)
Checking for existing v1.3 tables in v1.2 and fixing bigints...
Incorporating additional changes...
Loading sites/all/modules/tripal/tripal_core/chado_schema/default_schema-1.2-1.3-diff.sql...
WD tripal_core: FAILED. Line  1926, 0                                [error]
SQLSTATE[42804]: Datatype mismatch: 7 ERROR:  operator class
"int4_ops" does not accept data type bigint:
ALTER TABLE featuregroup
    ALTER featuregroup_id TYPE bigint,
    ALTER subject_id TYPE bigint,
    ALTER object_id TYPE bigint,
    ALTER group_id TYPE bigint,
    ALTER srcfeature_id TYPE bigint,
    ALTER fmin TYPE bigint,
    ALTER fmax TYPE bigint;
```
I see you have an issue tripal/tripal#896 that is a bit similar, but not really. Do we need to change some datatype to get past this? What do you think?

----

```
bradfordcondon commented 2 hours ago • 
looks like your index is blocking altering the type of that field. Drop the index and try again. if you have additional errors that look like that, you might need to drop (and probably want to re-add, after migration) all your indices of that type.
```

----
Thanks @bradfordcondon 

I feel like I am asking a question that I should know the answer to: What index should I be dropping? From that error message I don't see which index. I did query my db to find all indexes for featuregroup. Which should I drop? And then how do I rebuild?

```
drupal=# select indexname from pg_indexes where indexname like '%featuregroup%';
     indexname
-------------------
 featuregroup_c1
 featuregroup_pkey
 featuregroup_idx1
 featuregroup_idx2
 featuregroup_idx3
 featuregroup_idx4
 featuregroup_idx5
 featuregroup_idx6
(8 rows)
```

Another thing, I don't have a featuregroup table. Should I? Is my db totally hosed?

```
drupal=# select * from featuregroup limit 1;
ERROR:  relation "featuregroup" does not exist
LINE 1: select * from featuregroup limit 1;
                      ^
```

but I do have a feature_grp table.

Again Thank you!
Sofia

**Comment by spficklin** [comment]
Hi @srobb1 .  The `featuregroup` table is in the `frange` schema.  Tripal doesn't use, but it is part of Chado.  Do you see it  there on your site?

This problem does look to be index related.  The `int4_ops` is the class that PostgreSQL uses to search through an index.  So an `int8_ops` would be more appropriate for a `bigint` column which is what all of the ID columns in Chado tables get changed to in the upgrade file for Chado v1.3.  

So, you're getting this error on the `featuregroup` table which is not the first table to have the `int` changed to `bigint`, so I'm a bit baffled why you're getting it on that table.   Can you take a look at the indexes on that table (provided you see it in the `frange` schema) and see what you have?  Here are the indexes that I have for that table:
```
bingroup_boxrange 
featuregroup_c1 
featuregroup_idx1 
featuregroup_idx2 
featuregroup_idx3 
featuregroup_idx4 
featuregroup_idx5 
featuregroup_idx6 
featuregroup_pkey
```

**Comment by srobb1** [comment]
Hi @spficklin 

I am happy to see that I do have the `featuregroup` table in the `frange` schema :)

Here are my indexes for this table.
```
drupal=# select indexname from pg_indexes where tablename = 'featuregroup';
     indexname
-------------------
 featuregroup_c1
 featuregroup_pkey
 featuregroup_idx1
 featuregroup_idx2
 featuregroup_idx3
 featuregroup_idx4
 featuregroup_idx5
 featuregroup_idx6
(8 rows)
```

I am missing the bingroup_boxrange for `featuregroup` that you have listed.


Sofia

**Comment by spficklin** [comment]
So, it's very interesting that you have this problem only on the `featuregroup` table (so far). I don't understand why an index class change would be problematic for this table but not others.  One idea is to do as @bradfordcondon suggests and remove the indexes and try again.  Before you do so, you can dump the table SQL from postgres. That way if you need to restore the indexes afterwards you have easy access to the SQL to do it.

**Comment by srobb1** [comment]
We have this worked out. @bradfordcondon was correct. We started with deleting the feauregroup_idx*. The next error was with the index for cvtermpath. We deleted those and got the next table. So we deleted all indexes for all chado tables.

We followed @bradfordcondon example and generated drop index statements with this 
```
set search_path=public,so,frange,genetic_code,chado;

select 'drop index ' || indexname || ';'
from pg_indexes where indexname like '%idx%' AND schemaname = 'chado';
```

Before we dropped the indices as @spficklin suggested, we dumped the sql in case we need to recreate any index that doesn't automagically get recreated.

we could then sucessfully upgrade chado 1.2 -> 1.3

Thanks for the help!
Sofia
